src/types.ts

```diff
@@ -167,15 +175,18 @@ export interface MarkersProps {
   style: CSSProperties;
 }
 
-type Dimensions = Omit<PlotProps, 'colorScheme' | 'children'>;
 export interface PlotObjectType {
   axes: AxisProps[];
   series: Array<
     | ({ type: 'line' } & LineSeriesProps)
     | ({ type: 'scatter' } & ScatterSeriesProps)
   >;
   legend?: LegendProps;
-  dimensions: Dimensions;
+  dimensions?: Pick<PlotProps, 'width' | 'height' | 'margin'>;
+  svg?: Pick<
+    PlotProps,
+    'plotViewportStyle' | 'svgStyle' | 'seriesViewportStyle'
+  > & { class?: string; id?: string };
```

Comment:
Here, I'd also rename `svgStyle` to `style` and use `className` because that's more "react"

Reply:
That was part of the discussion that we have with @stropitek, to have the object definition "framework free" (svgStyle I think that has sense to rename anyway)

Reply:
even outside of a framework, the DOM property is called `className`

Reply:
I also prefer `svg.style` over `svg.svgStyle`
I'm good with `className` didn't think about how the property was named outside of react